rust/core-lib/src/file.rs

```diff
@@ -172,6 +190,79 @@ impl FileManager {
         }
         Ok(())
     }
+
+    #[cfg(feature = "notify")]
+    pub fn update_current_position_in_tail(
+        &mut self,
+        buffer_id: BufferId,
+    ) -> Result<bool, FileError> {
+        if let Some(file_info) = self.file_info.get_mut(&buffer_id) {
+            let path = &file_info.path.to_owned();
+
+            let mut f = File::open(path).map_err(|e| FileError::Io(e, path.to_owned()))?;
+            let end_position =
+                f.seek(SeekFrom::End(0)).map_err(|e| FileError::Io(e, path.to_owned()))?;
+
+            file_info.tail_position = Some(end_position);
+            self.watcher.unwatch(path, OPEN_FILE_EVENT_TOKEN);
+            self.watcher.watch(path, false, OPEN_FILE_TAIL_EVENT_TOKEN);
```

Comment:
I don't know inotify too well, but do you always need to do this and not just when you didn't have tailing enabled prefviously?

Reply:
Tailing is enabled only when user clicks "Tail" option under "Debug" menu. Just because a file is opened does not mean it will always be tailed. `disableTailing` is inverse of this. When tailing is disabled, the state goes back to the way it was before we started tailing.

Reply:
Because when a file is opened, doesnt mean it will always be tailed. This is inverse of `disableTailing` which resets the state to the way it was when file was opened.

Reply:
Yes, but wouldn't it be sufficient to only call this when we actually enable/disable tailing then instead of calling this every time we update the position?

Reply:
This method is only called when user enables tailing. Not every time file is updated.

Reply:
Oh, got it, sorry.